ts/README.md

```diff
@@ -0,0 +1,371 @@
+          this.observableMethod()
+            .subscribe(
+              (respone: any) => {
+                this.secondObservableMethod(response)
+                  .subscribe(
+                    () => {
+                      observer.complete();
+                    }
+                  );
+              }
+            );
+        });
+      }
+      ...
+    }
+
+    // good
+    class someClass {
+      ...
+      foo(): Observably<any> {
+        return this.observableMethod()
+          .pipe(
+            concatMap((respone: any) => this.secondObservableMethod(response))
+          );
+      };
+      ...
+    }
+    ```
+  <a name="async-data--wrapping-promises"></a><a name="4.3"></a>
+  - [4.3](#async-data--wrapping-promises) Avoid using promises and wrap them into observables and rxjs:
+    ```ts
+    // bad
+    class someClass {
+      ...
+      foo(): any {
+        return this.contractsService
+        .getContract(ContractsList.ProfileToken)
+        .methods
+        .getUserProfileTokenCount()
+        .call()
+        .then()
+        .catch()
+      };
+      ...
+    }
+
+    // good
+    class someClass {
+      ...
+      foo(): Observably<any> {
+        return from(this.contractsService
+          .getContract(ContractsList.ProfileToken)
+          .methods
+          .getUserProfileTokenCount()
+          .call()
+        )
+          .pipe(
+            catchError((error: any) => throwError(error))
+          )
+      }
+      ...
+    }
+    ```
+
+## Functions
+  <a name="functions--only-arrow-functions"></a><a name="5.1"></a>
+  - [5.1](#functions--only-arrow-functions) Use only arrow functions, avoid old plain construction:
+    ```ts
+    // bad
+    class someClass {
+      ...
+      const filteredArray: Array<any> = someArray.filter(function(item: any) {return item === 1});
+      ...
+    }
+
+    // good
+    class someClass {
+      ...
+      const filteredArray: Array<any> = someArray.filter((item: any) => item === 1);
+      ...
+    }
+    ```
+
+  <a name="functions--alignments"></a><a name="5.2"></a>
+  - [5.2](#functions--alignments) Align method's arguments (when they are over 140 letters) and chains into one column with symmetry:
+    ```ts
+    // bad
+    class someClass {
+      ...
+      this.service.method()
+      .pipe()
+      .subscribe();
+      ...
+    }
+
+    // good
+    class someClass {
+      ...
+      this.service
+        .method()
+        .pipe()
+        .subscribe();
+      ...
+    }
+    ```
+
+## Imports
+  <a name="imports--global-imports"></a><a name="6.1"></a>
+  - [6.1](#imports--global-imports) Avoid global imports from library, specific modules imports are more preferable:
+    ```ts
+    // bad
+    import { MatInputModule } from '@angular/material';
+    import { Observable } from 'rxjs/Rx';
+
+    // good
+    import { MatInputModule } from '@angular/material/input';
+    import { Observable } from 'rxjs';
```

Comment:
The observable import example is a bit confusing here, please remove or change to another one.

Reply:
@tracetoio-jingyi updated, please check.